src/averageAges.js

```diff
@@ -36,9 +39,16 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
-function calculateWomenAverageAge(people, withChildren) {
-  // write code here
-}
+const calculateWomenAverageAge = (people, withChildren) => {
+  const woman = withChildren ? people.filter(person =>
+    person.sex === 'f' && people.some(child => child.mother === person.name))
+    : people.filter(person => person.sex === 'f');
+
+  const sumOfAge = woman.reduce((sum, person) =>
+    sum + (person.died - person.born), 0);
+
+  return sumOfAge / woman.length;
```

Comment:
you have used the same logic for men. It would be a good idea to make a function for repeated logic